raiden-ts/tests/e2e/e2e.spec.ts

```diff
@@ -66,68 +71,262 @@ function getToken(): string {
+      BigNumber.from(amount - withdraw_amount - amount2),
+    );
+
+    /*
+     * Close channel with PC
+     */
+    await expect(raiden.closeChannel(getToken(), partner1)).resolves.toMatch('0x');
+
+    /*
+     * Deposit and withdraw between two LCs
+     *
+     * 50 tokens are left from withdraw above, so no need to mint.
+     */
+    const withdrawAmount2 = 10;
+    await expect(
+      raiden.depositChannel(getToken(), raiden2.address, withdraw_amount),
+    ).resolves.toMatch('0x');
+    await expect(getChannelCapacity(raiden, raiden2.address)).resolves.toStrictEqual(
+      BigNumber.from(withdraw_amount),
+    );
+
+    await expect(
+      raiden.withdrawChannel(getToken(), raiden2.address, withdrawAmount2),
+    ).resolves.toMatch('0x');
+    await expect(getChannelCapacity(raiden, raiden2.address)).resolves.toStrictEqual(
+      BigNumber.from(withdraw_amount - withdrawAmount2),
+    );
+
+    /*
+     * Send mediated payment with LC as mediator
+     *
+     * For this we need to enable medition in LC2
+     */
+    raiden2.updateConfig({
+      caps: {
+        [Capabilities.MEDIATE]: 1,
+      },
     });
+    await expect(raiden2.mint(getToken(), amount)).resolves.toMatch('0x');
+    await expect(raiden2.getTokenBalance(getToken())).resolves.toStrictEqual(
+      BigNumber.from(amount),
+    );
+    await expect(raiden2.openChannel(getToken(), partner2, { deposit: amount })).resolves.toMatch(
+      '0x',
+    );
+    await expect(getChannelCapacity(raiden2, partner2)).resolves.toStrictEqual(
+      BigNumber.from(amount),
+    );
+
+    // PC nodes need a long time to send capacity updates
+    await wait(10000);
+    const sendAmount = 35;
+
+    const routes2: RaidenPaths = await raiden.findRoutes(getToken(), partner2, sendAmount);
+    expect(routes2).toMatchObject([{ fee: BigNumber.from(0), path: [raiden2.address, partner2] }]);
+
+    key = await raiden.transfer(getToken(), partner2, sendAmount, { paths: routes2 });
+    await raiden.waitTransfer(key);
+
+    await expect(getChannelCapacity(raiden, raiden2.address)).resolves.toStrictEqual(
+      BigNumber.from(withdraw_amount - withdrawAmount2 - sendAmount),
+    );
+    await expect(getChannelCapacity(raiden2, partner2)).resolves.toStrictEqual(
+      BigNumber.from(amount - sendAmount),
+    );
+
+    /*
+     * Settle channel of LC1
+     *
+     * Need to wait until the settlement window is over.
+     */
+    await waitUntilSettleable(raiden, partner1);
+    await expect(raiden.settleChannel(getToken(), partner1)).resolves.toMatch('0x');
+
+    /*
+     * Withdraw from UDC
+     */
+    const udcWithdrawAmount = 10;
+    await expect(
+      raiden.getTokenBalance(process.env.SVT_TOKEN_ADDRESS as string),
+    ).resolves.toStrictEqual(BigNumber.from(0));
+
+    await expect(raiden.planUDCWithdraw(udcWithdrawAmount)).resolves.toMatch('0x');
+    await expect(raiden.withdrawFromUDC(udcWithdrawAmount)).resolves.toMatch('0x');
```

Comment:
@andrevmatos This API is a bit counter-intuitive for me. I expected it to wait until it can to the withdraw and execute it, but it fails when the withdraw plan is not "ready" yet.

Reply:
Addition: It's kind of the same for the `settle` call. I think we should either provide a `waitUntilSettleable` method or make `settleChannel` itself wait.

Reply:
Thanks for fixing this!